pkg/userdata/bottlerocket/etcd_init.go

```diff
@@ -18,10 +17,10 @@ runcmd: "{{ .EtcdadmInitCommand }}"
 `
 
 // NewInitEtcdPlane returns the user data string to be used on a etcd instance.
-func NewInitEtcdPlane(input *userdata.EtcdPlaneInput, config bootstrapv1alpha3.BottlerocketConfig, log logr.Logger) ([]byte, error) {
+func NewInitEtcdPlane(input *userdata.EtcdPlaneInput, config userdata.Config, log logr.Logger) ([]byte, error) {
```

Comment:
At this point, why not just pass the whole `bootstrapv1alpha3.EtcdadmConfig`?
If we want a separate struct, then I would vote to make it independent from the api structs. That would have the added benefit of being able to extract this code as a separate module that can be reused in difference controllers